opt.d/lib.sh

```diff
@@ -1,74 +1,151 @@
+    local sha_sum_url="${tool_url}.sha256sum"
+    local sha_sum_sig_url="${sha_sum_url}.asc"
+
+    if [ ! -f "$tool_tarball" ]; then
+        curl -L -f -o "$tool_tarball" "$tool_url"
+        if [ $? -ne 0 ]; then echo "curl failed for tarball $tool_tarball"; exit 1; fi
+    fi
+
+    if [ ! -f "$sha_sum_file_name" ]; then
+        curl -L -f -o "$sha_sum_file_name" "$sha_sum_url"
+        if [ $? -ne 0 ]; then echo "curl failed for sha256sum file $sha_sum_file_name"; exit 1; fi
+    fi
+
+    if [ ! -f "$sha_sum_sig_file_name" ]; then
+        curl -L -f -o "$sha_sum_sig_file_name" "$sha_sum_sig_url"
+        if [ $? -ne 0 ]; then echo "curl failed for sha256sum signature $sha_sum_sig_file_name"; exit 1; fi
+    fi
+
+    gpg --keyserver "hkps://$keyserver" --recv-keys $tool_signing_keys || { echo "gpg recv-keys failed for $tool_signing_keys"; exit 1; }
+
+    gpg --verify "$sha_sum_sig_file_name" "$sha_sum_file_name" || { echo "gpg verify failed for checksum file $sha_sum_file_name"; exit 1; }
+
+    grep " $tool_tarball\$" "$sha_sum_file_name" > "${sha_sum_file_name}.for_verify" 
+    if [ ! -s "${sha_sum_file_name}.for_verify" ]; then
+        echo "ERROR: Could not find checksum for '$tool_tarball' in '$sha_sum_file_name'"
+        echo "Contents of '$sha_sum_file_name':"
+        cat "$sha_sum_file_name"
+        rm -f "${sha_sum_file_name}.for_verify"
+        exit 1
+    fi
+
+    sha256sum --status -c "${sha_sum_file_name}.for_verify"
+    local checksum_rc=$?
+    rm -f "${sha_sum_file_name}.for_verify"
+
+    if [ $checksum_rc -ne 0 ]; then
+        echo "SHA256 CHECKSUM MISMATCH for $tool_tarball!"
+        echo "Expected sum from $sha_sum_file_name:"
+        grep " $tool_tarball\$" "$sha_sum_file_name"
+        echo "Calculated sum:"
+        sha256sum "$tool_tarball"
+        exit 1
+    fi
+
+    test -d "$tool_dir" || { echo "Extracting $tool_tarball..."; tar zxf "$tool_tarball" || { echo "tar extract failed for $tool_tarball"; exit 1; } }
+    rm -f "$sha_sum_file_name" "$sha_sum_sig_file_name"
+
+    cd "$tool_dir" || { echo "cd to $tool_dir failed"; exit 1; }
+}
+
+SetupStandardBuild() {
+    if [ ! -f "$tool_tarball" ]; then
+        curl -L -f -o "$tool_tarball" "$tool_url"
+        if [ $? -ne 0 ]; then echo "curl failed for tarball $tool_tarball"; exit 1; fi
+    fi
+
+    if [ ! -f "$tool_sig" ]; then
+        curl -L -f -o "$tool_sig" "$tool_sig_url"
+        if [ $? -ne 0 ]; then echo "curl failed for signature $tool_sig"; exit 1; fi
+    fi
+
+    gpg --keyserver "hkps://$keyserver" --recv-keys $tool_signing_keys || { echo "gpg recv-keys failed for $tool_signing_keys"; exit 1; }
+
+    gpg --verify "$tool_sig" "$tool_tarball" || { echo "gpg verify failed for $tool"; exit 1; }
+
+    test -d "$tool_dir" || { echo "Extracting $tool_tarball..."; tar zxf "$tool_tarball" || { echo "tar extract failed for $tool_tarball"; exit 1; } }
+    cd "$tool_dir" || { echo "cd to $tool_dir failed"; exit 1; }
```

Comment:
The code style and error messages are much clearer now to someone less familiar with shell scripts, thanks for doing this 🙏